ext/src/string_extension.cr

```diff
@@ -0,0 +1,71 @@
+# exact port from ActiveSupport in Ruby:
+# https://github.com/rails/rails/blob/42b0c942520e59399d70c2170253aa5275a42af1/activesupport/lib/active_support/core_ext/object/blank.rb#L101-L119
+class String
+  BLANK_RE = /\A[[:space:]]*\z/
+
+  # A string is blank if it's empty or contains whitespaces only:
+  #
+  #   ''.blank?       # => true
+  #   '   '.blank?    # => true
+  #   "\t\n\r".blank? # => true
+  #   ' blah '.blank? # => false
+  #
+  # Unicode whitespace is supported:
+  #
+  #   "\u00a0".blank? # => true
+  #
+  # @return [true, false]
+  def slow_blank?
+    BLANK_RE === self
+  end
+
+  def blank_as?
+    return true if self.nil? || self.size == 0
+    self.each_char do |char|
+      case char
+      when "\u0009",
```

Comment:
As a side note, this will fail for non-UTF-8 Ruby strings. Basically, you can't use crystal String for this because a Crystal String is assumed to be UTF-8, and iterating its char will assume this or otherwise fail.

The way to correctly do this is by translating the C code to Crystal. You'll end up with C code with Crystal/Ruby syntax. I'm not sure it's something good, because Crystal also comes with a GC and a heavy runtime.

I honestly would prefer to use C or Rust to write Ruby extensions, in my opinion Crystal is not a good fit for this. But that's just my opinion, maybe if we push this idea forward in the future Crystal will be more suitable for this, not sure.